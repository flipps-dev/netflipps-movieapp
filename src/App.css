@import url("https://fonts.googleapis.com/css2?family=Readex+Pro:wght@300;400;600;700;900&display=swap");

:root {
  --primary: #071013;
  --secondary: #EB5160;
  --btn-secondary: #B7999C;
  --white: #FCFCFC;
}

body {
  font-family: "Readex Pro", sans-serif;
  padding: 0;
  margin: 0;
}

.container {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width: 768px) {
  .container {
    width: 750px;
  }
}

@media (min-width: 992px) {
  .container {
    width: 970px;
  }
}

@media (min-width: 1200px) {
  .container {
    width: 1170px;
  }
}

*,
*:before,
*:after {
  box-sizing: border-box;
}

.btn {
  padding: 10px 15px;
  background-color: var(--secondary);
  color: var(--primary);
  border-radius: 5px;
  text-transform: uppercase;
  font-weight: 700;
  display: inline-block;
  border: none;
  font-size: 1rem;
  transition: all 0.3s ease;
  line-height: 1.1;
}

.btn:hover {
  opacity: 1;
  cursor: pointer;
}

.btn:disabled {
  pointer-events: none;
  opacity: 0.5;
}

.btn-secondary {
  padding: 10px 15px;
  background-color: var(--btn-secondary);
}

.btn-secondary:hover {
  background-color: var(--secondary);
  transition: all 0.3s ease;
}

input:focus {
  outline: none;
}

/* Header Styles */
header {
  background-color: var(--primary);
}

header .inner-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 0;
}

header a {
  color: var(--white);
  font-size: 1.25rem;
  text-decoration: none;
  transition: all 0.3s ease;
}

header a:hover {
  opacity: 0.7;
}

header .brand a {
  font-weight: 700;
  font-size: 1.5rem;
}

header .brand h1 {
  color: var(--white);
}

header .nav-links {
  padding: 0;
  margin: 0;
  list-style: none;
  display: flex;
  align-items: center;
}

header .nav-links li {
  display: inline-block;
  margin-right: 30px;
}

header .nav-links li:last-child {
  margin-right: 0;
}

header .btn-main {
  transition: all 0.4s;
  cursor: pointer;
}

header .btn-main span {
  cursor: pointer;
  display: inline-block;
  position: relative;
  transition: 0.4s;
}

header .btn-main span::after {
  content: '\002b';
  position: absolute;
  opacity: 0;
  top: 0;
  right: -20px;
  transition: 0.4s;
}

header .btn-main:hover span {
  padding-right: 25px;
}

header .btn-main:hover span:after {
  opacity: 1;
  right: 0;
}

/* Search */
.add-page {
  width: 100vw;
}
.add-content {
  padding: 50px 0;
  max-width: 600px;
  margin: 0 auto;
}

.add-content .input-wrapper input {
  width: 100%;
  background-color: rgb(223, 223, 223);
  padding: 10px;
  border-radius: 5px;
  border: none;
  color: var(--primary);
  font-size: 1.25rem;
}

.results {
  padding: 0;
  margin: 0;
  margin-top: 20px;
}

.results li {
  list-style: none;
}

.result-card {
  display: flex;
  margin-bottom: 20px;
}

.result-card .poster-wrapper img,
.filler-poster {
  width: 75px;
  height: 113px;
  background-color: #dbdada;
  border-radius: 5px;
  margin-right: 15px;
  display: block;
  color: transparent;
}

.result-card .info {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.result-card .info .title {
  font-size: 1.25rem;
  color: var(--primary);
  font-weight: 600;
  margin: 0;
}

.result-card .info .rate {
  font-size: 1rem;
  color: var(--primary);
  font-weight: 400;
  margin: 0;
}

.result-card .info .rate span {
  font-size: 1rem;
  color: var(--primary);
  font-weight: 600;
  margin: 0;
}

.result-card .info .release-date {
  font-size: 1.25rem;
  font-weight: 300;
  color: var(--primary);
  margin: 0;
}

.result-card .controls button {
  margin-right: 10px;
}

/* Movie Page */
.movie-page {
  padding: 30px 0;
}

.movie-page .header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 30px;
}

.count-pill {
  display: block;
  background-color: var(--secondary);
  padding: 5px 15px;
  font-size: 1.125rem;
  font-weight: 600;
  border-radius: 999px;
  color: var(--primary);
}

.movie-page .heading {
  color: var(--primary);
  margin: 0;
}

.movie-grid {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 30px;
}

@media (min-width: 992px) {
  .movie-grid {
    grid-template-columns: repeat(5, 1fr);
  }
}

.movie-card {
  width: 100%;
  border-radius: 5px;
  overflow: hidden;
  position: relative;
  border: 0;
  display: block;
}

.movie-card img {
  width: 100%;
  display: block;
}

.movie-card .overlay {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 2px solid transparent;
  transition: all 0.3s ease;
}

.movie-card:hover .overlay {
  border: 3px solid var(--secondary);
}

.movie-card .inner-card-controls {
  display: inline;
  background-color: red;
  position: absolute;
  bottom: 20px;
  left: 50%;
  transform: translateX(-50%);
  background: rgba(0, 0, 0, 0.8);
  border-radius: 5px;
  padding: 3px;
  opacity: 0;
  transition: all 0.3s ease;
}

.movie-card:hover .inner-card-controls {
  opacity: 1;
}

.ctrl-btn {
  color: #fefefe;
  background-color: transparent;
  border: none;
  transition: all 0.3s ease;
  font-size: 1.25rem;
  padding: 5px;
  margin: 0;
}

.ctrl-btn:hover {
  color: var(--secondary);
  cursor: pointer;
}

.movie-page .no-movies {
  color: #c4c4c4;
  font-size: 3rem;
  text-align: center;
}
