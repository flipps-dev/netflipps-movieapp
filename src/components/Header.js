import React from "react";
import { Link } from "react-router-dom";

export const Header = () => {
  return (
    <header>
      <div className="container">
        <div className="inner-content">
          <div className="brand">
            <h1 id="logo">Netflipps</h1>
          </div>
          <ul className="nav-links">
            <li id="li-toWatch">
              <Link to="/" id="btn-toWatch" className="btn btn-secondary">
                Da Vedere
              </Link>
            </li>

            <li id="li-seen">
              <Link to="/watched" id="btn-seen" className="btn btn-secondary">
                Visti
              </Link>
            </li>

            <li id="li-add">
              <Link to="/add" id="btn-add" className="btn btn-main">
                <span>Aggiungi </span>
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </header>
  );
};
