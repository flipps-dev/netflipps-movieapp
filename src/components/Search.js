import React, { useState } from "react";
import { SearchResult } from "./SearchResult";

export const Search = () => {
  
  const [query, setQuery] = useState("");
  const [results, setResults] = useState([]);

  const onChange = (e) => {
    e.preventDefault();

    setQuery(e.target.value);

    const URL_MOVIE = `https://api.themoviedb.org/3/search/movie?api_key=b91fd3bca472e98a57f91748abdafe02&language=it-IT&page=1&include_adult=false&query=${e.target.value}`;

    const URL_TV = `https://api.themoviedb.org/3/search/tv?api_key=b91fd3bca472e98a57f91748abdafe02&language=it-IT&page=1&include_adult=false&query=${e.target.value}`;
   
    fetch(URL_MOVIE)
      .then((res) => res.json())
      .then((data) => {
        if (!data.errors) {
          setResults(data.results);
        } else {
          setResults([]);
        }
        return fetch(URL_TV);
      }).then((res) => res.json())
      .then((data) => {
        if (!data.errors) {
          setResults(data.results);
        } else {
          setResults([]);
        }
      });


  };

  return (
    <div className="add-page">
      <div className="container">
        <div className="add-content">
          <div className="input-wrapper">
            <input
              type="text"
              placeholder="Cerca qualcosa di bello"
              value={query}
              onChange={onChange}
            />
          </div>

          {results.length > 0 && (
            <ul className="results">
              {results.map((movie) => (
                <li key={movie.id}>
                  <SearchResult movie={movie} />
                </li>
              ))}
            </ul>
          )}
        </div>
      </div>
    </div>
  );
};
