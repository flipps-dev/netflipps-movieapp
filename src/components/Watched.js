import React, { useContext } from "react";
import { GlobalContext } from "../context/GlobalState";
import { MovieCard } from "./MovieCard";

export const Watched = () => {
  const { watched } = useContext(GlobalContext);

  return (
    <div className="movie-page">
      <div className="container">
        <div className="header">
          <h1 className="heading">Visti</h1>

          <span className="count-pill">
            {watched.length} {watched.length === 1 ? "Elemento" : "Elementi"}
          </span>
        </div>

        {watched.length > 0 ? (
          <div className="movie-grid">
            {watched.map((movie) => (
              <MovieCard movie={movie} key={movie.id} type="watched" />
            ))}
          </div>
        ) : (
          <h2 className="no-movies">Non hai nulla nella lista, aggiungi qualcosa <i className="far fa-frown no-movies"></i></h2>
        )}
        
      </div>
    </div>
  );
};
