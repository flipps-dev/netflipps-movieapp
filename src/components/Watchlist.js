import React, { useContext } from "react";
import { GlobalContext } from "../context/GlobalState";
import { MovieCard } from "./MovieCard";

export const Watchlist = () => {
  const { watchlist } = useContext(GlobalContext);

  return (
    <div className="movie-page">
      <div className="container">
        <div className="header">
          <h1 className="heading">Da Vedere</h1>

          <span className="count-pill">
            {watchlist.length} {watchlist.length === 1 ? "Elemento" : "Elementi"}
          </span>
        </div>

        {watchlist.length > 0 ? (
          <div className="movie-grid">
            {watchlist.map((movie) => (
              <MovieCard movie={movie} key={movie.id} type="watchlist" />
            ))}
          </div>
        ) : (
          <h2 className="no-movies">Non hai nulla nella lista, aggiungi qualcosa <i className="far fa-frown no-movies"></i></h2>
        )}
        
      </div>
    </div>
  );
};
