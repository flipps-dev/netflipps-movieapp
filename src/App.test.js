import { render, screen, getByTestId } from '@testing-library/react';
import App from './App';
import { mount, shallow } from 'enzyme';
import { Search } from './components/Search';

//pulisco i test cookie
beforeAll(() => {
  document.cookie = '';
});

describe('AppComponent', () => {
  test("dovrebbe renderizzare correttamente in `debug` mode", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
})

// testo il componente Header
describe("Header Testing", () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />);
    //console.log(wrapper.debug());
  });

  afterEach(() => {
    wrapper.unmount();
 });

  test('il testo del logo Ã¨ `Netflipps`', () => {
    expect(wrapper.find("#logo").text()).toContain("Netflipps");
  });

  test('il bottone `Da Vedere` contiene il testo `Da Vedere`', () => {
    expect(wrapper.find(".nav-links").first().text()).toContain('Da Vedere');
  });

  test("il bottone `Visti` contiene il testo `Visti`", () => {
    expect(wrapper.find("#li-seen").text()).toContain("Visti");
  });

  test("il bottone `Aggiungi` contiene il testo `Aggiungi`", () => {
    expect(wrapper.find("#li-add").text()).toContain("Aggiungi");
  });

  test("includes link to `Da Vedere` page", () => {
    expect(wrapper.find("#btn-toWatch").first().props().to).toBe("/");
  });

  test("includes link to `Visti` page", () => {
    expect(wrapper.find("#btn-seen").first().props().to).toBe("/watched");
  });

  test("includes link to `Aggiungi` page", () => {
    expect(wrapper.find("#btn-add").first().props().to).toBe("/add");
  });
});

// testo il componente Searchbar
describe("Searchbar Testing", () => {
  
  test("renderizza correttamente un testo", () => {
    const strings = ["Ciao", "Mondo"];
    const component = shallow(<Search value={strings} />);
    expect(component).toMatchSnapshot();
  });


});



  

