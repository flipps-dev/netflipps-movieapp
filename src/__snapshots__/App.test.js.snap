// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent dovrebbe renderizzare correttamente in \`debug\` mode 1`] = `ShallowWrapper {}`;

exports[`Searchbar Testing renderizza correttamente un testo 1`] = `ShallowWrapper {}`;

exports[`Searchbar Testing stato iniziale della query Ã¨ una stringa 1`] = `ShallowWrapper {}`;
